common/fs/src/cache/mod.rs

```diff
@@ -73,64 +79,90 @@ impl<T: Default> FileSystem<T> {
                 if !path_cpy.exists() {
                     path_cpy.pop();
                 } else {
-                    fs.insert(&path_cpy, &mut |_, _| {});
+                    fs.insert(&path_cpy, &mut Vec::new());
```

Comment:
Is this right? I think it's equivalent... but I'm not 100% sure why we're ignoring the top level dirs